Bug Fixes and Reorder arrays

qmacAstanford opened the pull request:
Reorder R(NT,3) to R(3,NT) because I usually access x,y,z in succession.
Also make several Bug Fixes that probably only affect me.

qmacAstanford committed:
Added input to gitignore

qmacAstanford committed:
Removed lower case R

qmacAstanford committed:
Transpose R(NT,3) to R(3,NT) for speed, also U

Still need to transpose RT, UT, RS, US, RP, UP ...

qmacAstanford committed:
Transposed a lot of data arrays for speed

qmacAstanford committed:
Multiple Bug Fixes.  Fixed negetive phi problem

Bug Fix 1: Changed ABP to AB in MC_int_swap.f03 because ABP is not
updated.

Bug Fix 2: Set DESELF and etwist to 0 when not in use

Bug Fix 3: Addapt after the correct number of moves with interviening
exchanges.

Bug Fix 4: Removed turning moves on from tweak_parameters

qmacAstanford committed:
Cleared compile time errors from last commit

qmacAstanford committed:
Work in progress on Chemical Idennity change

qmacAstanford committed:
Added Maier Saupe (chain alignment) interaction

Code compiling by not tested (or even run).

Basically added an extra 5 fields (m=-2..2) and one energy term.

Parallel temperint in this new energy term is available.

Also split of the change in field calculation for chain sequence
moves to a separate subroutine: MC_int_chem.

qmacAstanford committed:
Input switches as integers to strins

Also did a number of little changes for setting up Maier Saupe

qmacAstanford committed:
Bug Fixes: Maier Saupe was not fully implemented

Fixed problems which became appairant from runtime errors and
performance problems.

qmacAstanford committed:
Bug Fixes: few bugs to git binding state to work.

qmacAstanford committed:
New Feature: WLC initial condition from Bruno

Commit:
changes to wlcsim_quinn.f03q

qmacAstanford committed:
Bug Fixes in progress for Maier Saupe

There are bus still known to exist.

brunobeltran commented on src/mc/MC_confine.f03:
```diff
@@ -32,42 +31,42 @@ subroutine MC_confine(confineType, LBox, RP, NT, IT1, IT2, ECon)
     do I = IT1,IT2
-        if(RP(I,3)<0.0_dp) then
+        if(RP(3,I)<0.0_dp) then
             ECon = 9990000.0_dp
```
We should make this large energy a constant. And while we're at it, we should do this for all "just put a big energy here" numbers.

brunobeltran commented on src/mc/MCsim.f03:
```diff
@@ -53,6 +53,7 @@ subroutine MCsim(wlc_p,wlc_d,NSTEP)
     Type(wlcsim_data), intent(inout) :: wlc_d     ! system allocated data
     integer DELTA             !Alexander polynomial evaluated at t = -1; used for knot checking
     real(dp) para(10)
+    integer m_plus3
```
comment that this "m" is a spherical harmonic quantum "m"

brunobeltran commented on src/mc/MC_int_swap.f03:
```diff
@@ -29,11 +30,13 @@ subroutine MC_int_swap(wlc_p,wlc_d,I1,I2,I3,I4)
-! Copy so I don't have to type wlc_p% everywhere
+!LOGICAL isA   ! The bead is of type A
+integer AminusB
+integer m_plus3
```
same

qmacAstanford replied:
fixed

brunobeltran commented on src/mc/MC_int_rep.f03:
```diff
@@ -33,12 +33,12 @@ subroutine MC_int_rep(wlc_p,wlc_d,I1,I2,forward)
-
-! Copy so I don't have to type wlc_p% everywhere
+real(dp) phi2(5)
+integer m_plus3
```
same

brunobeltran commented on src/mc/MC_int_chem.f03:
```diff
@@ -0,0 +1,163 @@
+integer NBinX(3)
+real(dp) temp
+real(dp) contribution 
+integer m_plus3
```
same

brunobeltran commented on src/mc/Y2calc.f03:
```diff
@@ -0,0 +1,24 @@
+
+real(dp), intent(in) :: u(3)
+real(dp), intent(out) :: Y2real(5)
+real(dp), parameter, dimension(5) :: coefficient = (/1.092548430592079, &
```
replace with symbolic values

qmacAstanford replied:
fixed as much as I can.  Replaced with locally defined PI.  Can't used pi from params as it isn't a parameter

qmacAstanford committed:
Bug Fix: now maier saupe energies add up

brunobeltran commented on src/mc/verifyEnergiesFromScratch.f03:
```diff
@@ -90,7 +90,7 @@ subroutine VerifyEnergiesFromScratch(wlc_p, wlc_d)
 
     ! --- Binding Energy ---
-    if(abs(wlc_d%EBind-wlc_d%DEBind).gt.0.00001) then
+    if(abs(wlc_d%EBind-wlc_d%DEBind) > 0.00001) then
```
replace with hardcoded float eps in params

brunobeltran commented on src/mc/verifyEnergiesFromScratch.f03:
```diff
@@ -99,7 +99,7 @@ subroutine VerifyEnergiesFromScratch(wlc_p, wlc_d)
 
     ! --- Elastic Energy ---
-    if(abs((wlc_d%EElas(1)+  wlc_d%EElas(2)+ wlc_d%EElas(3))-&
+    if(abs((wlc_d%EElas(1) +  wlc_d%EElas(2) + wlc_d%EElas(3))-&
```
replace with hardcoded float eps in params

qmacAstanford replied:
fixed

brunobeltran commented on src/mc/verifyEnergiesFromScratch.f03:
```diff
@@ -112,34 +112,44 @@ subroutine VerifyEnergiesFromScratch(wlc_p, wlc_d)
     if (wlc_p%field_int_on) then
         ! test to see if sum of changes are same as calculating from scratch
-        if(abs(wlc_d%EChi-wlc_d%DEChi).gt. 0.0001_dp) then
+        if(abs(wlc_d%EChi-wlc_d%DEChi) > 0.0001_dp) then
```
replace with hardcoded float eps in params

brunobeltran commented on src/mc/verifyEnergiesFromScratch.f03:
```diff
@@ -112,34 +112,44 @@ subroutine VerifyEnergiesFromScratch(wlc_p, wlc_d)
         wlc_d%EChi = wlc_d%DEChi
         wlc_d%x_chi = wlc_d%dx_chi
-        if(abs(wlc_d%ECouple-wlc_d%DECouple).gt. 0.0001_dp) then
+        if(abs(wlc_d%ECouple-wlc_d%DECouple) > 0.0001_dp) then
```
replace with hardcoded float eps in params

brunobeltran commented on src/mc/verifyEnergiesFromScratch.f03:
```diff
@@ -112,34 +112,44 @@ subroutine VerifyEnergiesFromScratch(wlc_p, wlc_d)
         wlc_d%ECouple = wlc_d%DECouple
         wlc_d%x_Couple = wlc_d%dx_couple
-        if(abs(wlc_d%EKap-wlc_d%DEKap).gt. 0.0001_dp) then
+        if(abs(wlc_d%EKap-wlc_d%DEKap) > 0.0001_dp) then
```
replace with hardcoded float eps in params

brunobeltran commented on src/mc/verifyEnergiesFromScratch.f03:
```diff
@@ -112,34 +112,44 @@ subroutine VerifyEnergiesFromScratch(wlc_p, wlc_d)
         wlc_d%x_Kap = wlc_d%dx_Kap
 
-        if(abs(wlc_d%EField-wlc_d%DEField).gt.0.00001) then
+        if(abs(wlc_d%EField-wlc_d%DEField) > 0.00001) then
```
replace with hardcoded float eps in params

brunobeltran commented on src/mc/verifyEnergiesFromScratch.f03:
```diff
@@ -112,34 +112,44 @@ subroutine VerifyEnergiesFromScratch(wlc_p, wlc_d)
         wlc_d%x_Field = wlc_d%dx_Field
+
+        if(wlc_p%chi_l2_on) then
+            if(abs(wlc_d%EMaiersaupe-wlc_d%deMaierSaupe) > 0.00001) then
```
replace with hardcoded float eps in params

brunobeltran commented on src/util/binning.f03:
```diff
@@ -171,25 +171,25 @@ recursive subroutine addBead(bin,R,NT,beadID)
         integer, intent(in) :: beadID
         integer binIndex
 
         real(dp), parameter :: eps = 0.00000001
```
should inherit from params instead

brunobeltran commented on src/util/getdim.f03:
```diff
@@ -35,18 +35,18 @@ subroutine getdim(N,NP,NT,R,RCOM,RCOMSQ,RGYRSQ,R2,R4,R6,DR)
+     X(1,:) = R(1,IB1:IBN)
+     X(2,:) = R(2,IB1:IBN)
+     X(3,:) = R(3,IB1:IBN)
+     RCOM(:,IP) = SUM(X,DIM = 1)/REAL(N)
```
dim should have changed

brunobeltran commented on src/wlcsim/MCparrll_mpi.f90:
```diff
@@ -183,6 +186,7 @@ subroutine replicaExchange(wlc_p, wlc_d)
     !wlc_p%para(3)  =cof(8)
+    wlc_p%chi_l2 = cof(9)
 
     if (abs(wlc_d%EChi-x(1)*CofOld(1)).gt.0.0000001_dp) then
```
immediately unpack arr and do things with names
for x and cof

brunobeltran replied:
params eps again

brunobeltran commented on src/wlcsim/MCparrll_mpi.f90:
```diff
@@ -198,6 +202,7 @@ subroutine replicaExchange(wlc_p, wlc_d)
    ! wlc_p%EElas(3) = wlc_p%EElas(3) + x(8)*(Cof(8)-CofOld(8))
+    wlc_d%eMaierSaupe    =wlc_d%eMaierSaupe    +x(9)*(Cof(5)-CofOld(5))
 
     if (abs(wlc_d%EChi-x(1)*Cof(1)).gt.0.000001_dp) then
```
params eps again

qmacAstanford committed:
Bug fix: US index order reversed in RKstep.f03

brunobeltran commented on src/mc/MC_move.f90:
```diff
@@ -203,50 +207,49 @@ subroutine MC_move(R,U,RP,UP,NT,NB,NP,IP,IB1,IB2,IT1,IT2,MCTYPE &
   !  ------begining testing---------
   if(.false.) then
       ! This is a code block for testing
-      if (abs(RP(IT1,1)-R(IT1,1)).gt.0.000001) then
```
params eps

brunobeltran commented on src/mc/MC_move.f90:
```diff
@@ -203,50 +207,49 @@ subroutine MC_move(R,U,RP,UP,NT,NB,NP,IP,IB1,IB2,IT1,IT2,MCTYPE &
   if(.false.) then
       ! This is a code block for testing
-      if (abs(RP(IT1,1)-R(IT1,1)).gt.0.000001) then
+      if (abs(RP(1,IT1)-R(1,IT1)).gt.0.000001) then
```
params eps

brunobeltran commented on src/mc/MC_move.f90:
```diff
@@ -203,50 +207,49 @@ subroutine MC_move(R,U,RP,UP,NT,NB,NP,IP,IB1,IB2,IT1,IT2,MCTYPE &
+          print*, RP(1,IT1), R(1,IT1)
           stop 1
       endif
-      if (abs(RP(IT2,1)-R(IT2,1)).gt.0.000001) then
```
params eps

brunobeltran commented on src/mc/MC_move.f90:
```diff
@@ -203,50 +207,49 @@ subroutine MC_move(R,U,RP,UP,NT,NB,NP,IP,IB1,IB2,IT1,IT2,MCTYPE &
+          d2 = (RP(1,IT1 + 1)-RP(1,IT1))**2 + &
+             (RP(2,IT1 + 1)-RP(2,IT1))**2 + &
+             (RP(3,IT1 + 1)-RP(3,IT1))**2
           if (abs(d1-d2).gt.0.000001) then
```
params eps

brunobeltran commented on src/mc/MC_move.f90:
```diff
@@ -752,16 +754,16 @@ subroutine test_equiv_forward(U,R,UP,RP,NT,IT1,IT2,RparaMag,RperpMag)
+         DRNew(2)*UP(2,IT2-1) + &
+         DRNew(3)*UP(3,IT2-1)
 if (abs(drOld(1)**2 + drOld(2)**2 + drOld(3)**2&
       -(drNew(1)**2 + drNew(2)**2 + drNew(3)**2)).gt.0.000001) then
```
params eps

brunobeltran commented on src/mc/MC_move.f90:
```diff
@@ -776,12 +778,12 @@ subroutine test_equiv_forward(U,R,UP,RP,NT,IT1,IT2,RparaMag,RperpMag)
+drPerpNew(3) = drNew(3)-drParNew*UP(3,IT2-1)
 
 if (abs(drPerpOld(1)**2 + drPerpOld(2)**2 + drPerpOld(3)**2 &
       -(drPerpNew(1)**2 + drPerpNew(2)**2 + drPerpNew(3)**2)).gt.0.000001_dp) then
```
params eps

brunobeltran commented on src/wlcsim/params.f03:
```diff
@@ -347,27 +360,31 @@ subroutine set_param_defaults(wlc_p,wlc_d)
+        wlc_p%CHI_l2 =0.0_dp ! don't use maier Saupe by default
         wlc_p%hA =0.0_dp  ! don't use weird artificial field by default
-        wlc_p%KAP =10.0_dp ! "fairly" incompressible --Quinn
+        wlc_p%KAP =0.0_dp ! "fairly" incompressible --Quinn
```
no longer "fairly" incompressible lol

brunobeltran commented on src/wlcsim/params.f03:
```diff
@@ -1220,7 +1260,7 @@ subroutine printDescription(wlc_p)
         print*, " volume fraction polymer =", wlc_p%Fpoly
         print*, " bead volume V = ", wlc_p%beadVolume
-        print*, " number of kuhn lengths between beads, eps ", wlc_p%eps
+        print*, " 1/number of kuhn lengths between beads, eps ", wlc_p%eps
```
nope, was correct as-was

brunobeltran commented on src/wlcsim/params.f03:
```diff
@@ -992,19 +1030,23 @@ subroutine initialize_wlcsim_data(wlc_d, wlc_p)
+        allocate(wlc_d%R(3,NT))
+        allocate(wlc_d%U(3,NT))
         if (wlc_p%codeName /= 'bruno' .OR. wlc_p%nInitMCSteps /= 0) then
-            allocate(wlc_d%RP(NT,3))
```
these can be pulled out of the if statement now actually, can we do that here? thanks!

qmacAstanford replied:
I always need RP but I don't know if you do when you are only doing bd.  I'll let you make that call.

brunobeltran reviewed (changes requested):
recommend cof and cofMtrx renamed to coefs in src/wlcsim/MCparrll_mpi.f90 and src/wlcsim/wlcsim_quinn.f03.

all changes minor, except one bug in BD. 

Accepted for merge with these changes.

brunobeltran committed:
BUGFIX: type from awk-based edits to stop_if_err

brunobeltran committed:
REFACTOR: confinment code/input style

fixed bug where randomWalkWithBoundary initial condition type could
create chains that were outside the boundary

qmacAstanford committed:
Organization: Split MC_move into subroutines

Put each in its own file.

brunobeltran committed:
BUGFIX: IN PROGRESS: initcond stuff refactored/wrong SIGMA!

qmacAstanford committed:
Implemented Bruno's Code review suggestions

Also updated input params (I have this gitignored) but I want an updated
one on the git which actually runs.

qmacAstanford committed:
New Feature: Super Reptation

Super Reptation is a mc move which removes a bead from one end of a
polymer and adds one to the other.  The chemical identities are changed
inbetween so that the AB field is only affected at the ends.

Also combined moveTypes and nMoveTypes as they are always the same.

Also did some slight reorganization of move arguements.

qmacAstanford committed:
Forgot to add.  See last commit.

brunobeltran commented:
I am able to compile the code using params::pi in definition of constants. We should clear this up then merge these changes! Super great work with all this cleanup and new code!

qmacAstanford committed:
Pi made a parameter, only defined once.
